cargo-pgx/src/templates/pgx-linker-script.sh

```diff
@@ -2,7 +2,17 @@
 # Auto-generated by pgx. You may edit this, or delete it to have a new one created.
 
 if [[ $CARGO_BIN_NAME == "sql-generator" ]]; then
-    gcc -Wl,-undefined,dynamic_lookup,-dynamic-list=$CARGO_MANIFEST_DIR/.cargo/pgx-dynamic-list.txt $@
-else 
+    if [[ $TARGET == "x86_64-apple-darwin" ]] || [[ $TARGET == "aarch64-apple-darwin" ]]; then
```

Comment:
can we just look for `*apple-darwin*` here?  I feel it'll be unlikely for different apple architectures to do different things and it'll let us support new apple ones in the future without thinking.

Reply:
Yeah I think I can do `~=`!

Reply:
I fixed this, on accident, by using `uname` since `$TARGET` was empty.  haha.